### PR TITLE
[Doc] add different pages for java/kotlin multiplatform

### PR DESCRIPTION
**Pull request by martinbonnin**
This PR moves the basic instructions to get started to 3 different pages depending if users want to get started with:

* Java
* Kotlin + Coroutines
* Multiplatform

It removes some of the initial complexity by only exposing what's needed for a single use case and not asking users to make choices (besides the initial platform choice obvisouly :)).

This PR also adds basic documentation about fragments and inline fragments.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,168 +66,42 @@ dependencies {
...
+* JDK 8+
+* iOS 13
+ 
 ## Contributing
 
 If you'd like to contribute, please see [Contributing.md](https://github.com/apollographql/apollo-android/blob/master/Contributing.md).
 
+## Additional resources
+
+- [#125, Fragmented Podcast](http://fragmentedpodcast.com/episodes/125/): Why's and How's about Apollo Android and the entire journey.
```
Comment (tasomaniac) [inline README.md:99]: Would be nice to also put a link to our slides. 🤘

File: docs/source/shared/add-query.mdx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+1. Create a directory for your GraphQL files: `src/main/graphql/com/example/`
```
Comment (tasomaniac) [inline docs/source/shared/add-query.mdx:1]: These shared files are not included in the index, is that intentional?

File: README.md (hunk 1 above)
Comment (martinbonnin) [reply README.md:100]: 💯 I just added them

File: docs/source/shared/add-query.mdx (hunk 2 above)
Comment (martinbonnin) [reply docs/source/shared/add-query.mdx:1]: Yep, they are shared between the Java/Kotlin/Multiplatform pages thanks to the `.mdx` magic

Comment (tasomaniac) [reply docs/source/shared/add-query.mdx:1]: Today I learned.

**Pull request merged**